motto/llm_gate.py

```diff
@@ -162,8 +164,11 @@ def llm(metta: MeTTa, *args):
         agent = MettaAgent(agent)
     if not isinstance(agent, Agent):
         raise TypeError(f"Agent {agent} should be of Agent type. Got {type(agent)}")
-    response = agent(msgs_atom if isinstance(agent, MettaAgent) else messages,
-                     functions)
+    if isinstance(agent, RetrievalAgent):
```

Comment:
I'd add `functions` to the retrieval agent call and didn't change llm-gate here.